src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs

```diff
@@ -12642,5 +12642,876 @@ public void ScopedReserved_Alias_Escaped()
+                //     public ref int Property1 => ref _field; // 1
+                Diagnostic(ErrorCode.ERR_RefReturnStructThis, "_field").WithLocation(5, 37));
+        }
+
+        [CombinatorialData]
+        [Theory]
+        public void UnscopedRefAttribute_Property_02(bool useCompilationReference)
+        {
+            var sourceA =
+@"using System.Diagnostics.CodeAnalysis;
+public struct S<T>
+{
+    private T _t;
+    public ref T P1 => throw null;
+    [UnscopedRef] public ref T P2 => ref _t;
+}";
+            var comp = CreateCompilation(new[] { sourceA, UnscopedRefAttributeDefinition });
+            comp.VerifyEmitDiagnostics();
+            var refA = AsReference(comp, useCompilationReference);
+
+            var sourceB =
+@"class Program
+{
+    static ref int F1()
+    {
+        var s = new S<int>();
+        return ref s.P1;
+    }
+    static ref int F2()
+    {
+        var s = new S<int>();
+        return ref s.P2; // 1
+    }
+}";
+            comp = CreateCompilation(sourceB, references: new[] { refA });
+            // https://github.com/dotnet/roslyn/issues/62791: Missing errors.
+            comp.VerifyEmitDiagnostics();
+        }
+
+        [WorkItem(62691, "https://github.com/dotnet/roslyn/issues/62691")]
+        [Fact]
+        public void UnscopedRefAttribute_RefRefStructParameter_01()
+        {
+            var source =
+@"using System.Diagnostics.CodeAnalysis;
+ref struct R { }
+class Program
+{
+    static ref R ReturnRefStructRef(bool b, ref R x, [UnscopedRef] ref R y)
+    {
+        if (b)
+            return ref x; // 1
+        else
+            return ref y;
+    }
+}";
+            // https://github.com/dotnet/roslyn/issues/62691: An error should be reported for 'return ref x;'.
+            var comp = CreateCompilation(new[] { source, UnscopedRefAttributeDefinition });
+            comp.VerifyDiagnostics();
+
+            var parameters = comp.GetMember<MethodSymbol>("Program.ReturnRefStructRef").Parameters;
+            VerifyParameterSymbol(parameters[1], "ref R x", RefKind.Ref, DeclarationScope.Unscoped); // https://github.com/dotnet/roslyn/issues/62691: Should be DeclarationScope.RefScoped.
+            VerifyParameterSymbol(parameters[2], "ref R y", RefKind.Ref, DeclarationScope.Unscoped);
+        }
+
+        [CombinatorialData]
+        [Theory]
+        public void UnscopedRefAttribute_RefRefStructParameter_02(bool useCompilationReference)
+        {
+            var sourceA =
+@"using System.Diagnostics.CodeAnalysis;
+public ref struct R<T>
+{
+    public ref T F;
+    public R(ref T t) { F = ref t; }
+}
+public class A<T>
+{
+    public ref T F1A(ref R<T> r1)
+    {
+        throw null;
+    }
+    public ref T F2A(scoped ref R<T> r2)
+    {
+        throw null;
+    }
+    public ref T F3A([UnscopedRef] ref R<T> r3)
+    {
+        return ref r3.F;
+    }
+    public ref T F4A([UnscopedRef] scoped ref R<T> r4)
```

Comment:
Missing an error here `[UnscopedRef] scoped`

Reply:
This can be addressed in a follow-up as well. We should have tests and errors for: [UnscopedRef] disallowed on `init` and constructors.
Also for follow-up, we should test `scoped` or [UnscopedRef] on record positional members
Any interesting extension scenarios? `[UnscopedRef] this`?

Reply:
1. Added
2. ...
3. ...
4. Extension method `this` parameters should be treated as regular parameters.
5. Overrides and interface implementations: `UnscopedRefAttribute_Overrides*`, `_Implementations*`, `_Delegates*`
6. Added